subprojects/plugins/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt

```diff
@@ -172,14 +172,10 @@ if [ "\$cygwin" = "true" -o "\$msys" = "true" ] ; then
     esac
 fi
 
-# Escape application args
-save () {
-    for i do printf %s\\\\n "\$i" | sed "s/'/'\\\\\\\\''/g;1s/^/'/;\\\$s/\\\$/' \\\\\\\\/" ; done
-    echo " "
-}
-APP_ARGS=`save "\$@"`
+ARGV=("\$@")
+eval set -- \$DEFAULT_JVM_OPTS
```

Comment:
eval is still used for `DEFAULT_JVM_OPTS` - this variable is not set by the user in the script - the values of this one are defined at build time using `applicationDefaultJvmArgs` array. The values are then sanitized and escaped properly before being set in the script, which prevents injecting and executing code here.
Alternatively, the escaping Gradle does should be revisited to avoid eval here as well and set the array the same way like below.